### PR TITLE
[DocTest] Doctest for Electra Pytorch

### PR DESCRIPTION
**Pull request by bhadreshpsavani**
# What does this PR do?
Adds Doctest fo Electra Pytorch

Issue: #16292

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@ydshieh

### CODE REVIEW

File: src/transformers/models/electra/modeling_electra.py [hunk 1]
```diff
@@ -1256,9 +1261,11 @@ def __init__(self, config):
     @add_start_docstrings_to_model_forward(ELECTRA_INPUTS_DOCSTRING.format("batch_size, sequence_length"))
     @add_code_sample_docstrings(
         processor_class=_TOKENIZER_FOR_DOC,
-        checkpoint=_CHECKPOINT_FOR_DOC,
+        checkpoint="bhadresh-savani/electra-base-discriminator-finetuned-conll03-english ",
```
Comment (ydshieh) [inline src/transformers/models/electra/modeling_electra.py:1264]: nice!

**Review by ydshieh — COMMENTED** [review]
Hi, @bhadreshpsavani 

It looks great, thank you for your work! I leave a few comments.
(I also saw you uploaded a checkpoint, very nice!)

**Comment by bhadreshpsavani** [comment]
Hi @ydshieh,
For pytorch its running fine,
only In case of tensorflow its giving issue,
I am sharing [notebook](https://colab.research.google.com/drive/1buzYMKgm3b8kfTcyVvnToJCFLqdnZDxM?usp=sharing) for reproducing the issue.

**Comment by ydshieh** [comment]
> Hi @ydshieh, The issue we discussed in Issue:16292 For PyTorch its running fine, only In case of TensorFlow its giving issue, I am sharing [notebook](https://colab.research.google.com/drive/1buzYMKgm3b8kfTcyVvnToJCFLqdnZDxM?usp=sharing) for reproducing the issue.

Hi, I am not able to open it. Could you make it public?

**Comment by bhadreshpsavani** [comment]
Hi @ydshieh,
I have updated the link

**Comment by ydshieh** [comment]
@bhadreshpsavani Thanks. It looks like PyTorch somehow made it to run, despite the indices are wrong and out of range.
It gave strange loss value thought.

I opened a PR to address this situation

https://github.com/huggingface/transformers/pull/16648. Let's wait it being reviewed and merged to proceed.

Thank you for your patience!

**Comment by bhadreshpsavani** [comment]
Hi @ydshieh,

I think when I used this code and rebase at the end I messed up the commits, is there any way to remove earlier commits from this PR?

**Comment by ydshieh** [comment]
Hi, @bhadreshpsavani 

I think close this PR and create a new one is the easiest way (considering the change in your PR is not so large).
Thank you for the effort.

BTW, I think you used `git merge` rather than `git rebase`. In general, `git rebase` could avoid this situation :-).

**Comment by bhadreshpsavani** [comment]
I followed this,
```
git checkout main  # or `master`, depends on your local clone
git fetch upstream
git pull upstream main  # Hugging Face `transformers` renamed the default branch to `main` recently
git checkout your_working_branch_for_this_sprint
git rebase main  # or `master`
```
In the end, there was a merge conflict so i use something like `git rebase --continue` after solving the conflict and sync the changes. It got messed up!

I need to be more careful!

**Comment by ydshieh** [comment]
> I followed this,
> 
> ```
> git checkout main  # or `master`, depends on your local clone
> git fetch upstream
> git pull upstream main  # Hugging Face `transformers` renamed the default branch to `main` recently
> git checkout your_working_branch_for_this_sprint
> git rebase main  # or `master`
> ```
> 
> In the end, there was a merge conflict so i use something like `git rebase --continue` after solving the conflict and sync the changes. It got messed up!
> 
> I need to be more careful!

OK, conflict is not easy thing.
